File: cli/internal/commands/status.go
package commands

import (
	"fmt"
	"net/http"
	"time"

	"github.com/spf13/cobra"
)

var healthChecks = []struct {
	Name string
	URL  string
}{
	{"otel-collector", "http://localhost:4318/v1/logs"},
	{"prometheus", "http://localhost:9090/-/healthy"},
	{"clickhouse", "http://localhost:8123/ping"},
	{"grafana", "http://localhost:3000/api/health"},
}

func isUp(url string) bool {
	client := http.Client{Timeout: 3 * time.Second}
	resp, err := client.Get(url)
	if err != nil {
		return false
	}
	defer resp.Body.Close()
	return resp.StatusCode < 500
}

func StatusCmd() *cobra.Command {
	return &cobra.Command{
		Use:   "status",
		Short: "Health-check the collector/prometheus/clickhouse/grafana stack",
		RunE: func(cmd *cobra.Command, args []string) error {
			fmt.Printf("%-16s %s\n", "SERVICE", "STATUS")
			for _, check := range healthChecks {
				status := "down"
				if isUp(check.URL) {
					status = "up"
				}
				fmt.Printf("%-16s %s\n", check.Name, status)
			}
			return nil
		},
	}
}

File: cli/src/agentobs/commands/status.py
from __future__ import annotations

import httpx
from rich.console import Console
from rich.table import Table

console = Console()

CHECKS = {
    "otel-collector": "http://localhost:4318/v1/logs",  # POST-only endpoint; any response = up
    "prometheus": "http://localhost:9090/-/healthy",
    "clickhouse": "http://localhost:8123/ping",
    "grafana": "http://localhost:3000/api/health",
}


def _is_up(url: str) -> bool:
    try:
        resp = httpx.get(url, timeout=3)
        return resp.status_code < 500
    except httpx.HTTPError:
        return False


def status() -> None:
    """Health-check the collector/prometheus/clickhouse/grafana stack."""
    table = Table()
    table.add_column("Service")
    table.add_column("Status")

    for name, url in CHECKS.items():
        up = _is_up(url)
        table.add_row(name, "[green]up[/green]" if up else "[red]down[/red]")

    console.print(table)
